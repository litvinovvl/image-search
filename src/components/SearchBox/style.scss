.search-wrapper {
  background-color: #353335;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  &.shrink {
    height: 80px;
    transition: all 1s ease-in-out;
  }

  .search-input {
    height: 40px;
    width: 500px;
    border-radius: 4px;
    border: 1px solid;
    padding: 0 16px;

    &:focus-visible {
      outline: 2px solid #3A71D5;
      outline-offset: -2px;
    }

    &::placeholder {
      color: #D5D8DC;
    }
  }

  .search-btn {
    margin: 0 10px;
    height: 42px;
    padding: 12px 20px;
    background-color: #3A71D5;
    color: #FFF;
    cursor: pointer;
    border: none;
    border-radius: 4px;
    font-weight: bold;

    &:hover {
      opacity: 0.9;
    }

    &:active {
      opacity: 0.7;
    }
  }
}
