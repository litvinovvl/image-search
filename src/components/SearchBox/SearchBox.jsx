import React, { useState } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import './style.scss';

const SearchBox = ({ isLoading, fullScreen, onSubmit }) => {
  const [query, setQuery] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    onSubmit(query);
  };

  const handleChange = ({ target: { value } }) => setQuery(value);

  return (
    <form onSubmit={handleSubmit} className={classnames('search-wrapper', { shrink: !fullScreen })}>
      <input
        className="search-input"
        placeholder="What images would ypu like to see on Pixabay?"
        onChange={handleChange}
      />
      <button type="submit" className="search-btn">
        {isLoading ? 'Searching' : 'Search'}
      </button>
    </form>
  );
};

SearchBox.defaultProps = {
  fullScreen: true,
  isLoading: false,
};

SearchBox.propTypes = {
  fullScreen: PropTypes.bool,
  isLoading: PropTypes.bool,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchBox;
