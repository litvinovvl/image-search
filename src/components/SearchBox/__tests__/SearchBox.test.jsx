import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from '../SearchBox';

const props = {
  fullScreen: true,
  onSubmit: jest.fn(),
};

describe('App', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render SearchBox properly', () => {
    const component = shallow(<SearchBox {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should render SearchBox properly when images loaded', () => {
    const component = shallow(<SearchBox {...props} fullScreen={false} />);

    expect(component).toMatchSnapshot();
  });

  it('should call onSubmit with entered text when user submits query', () => {
    const value = 'some query';
    const component = shallow(<SearchBox {...props} />);
    const input = component.find('input');

    input.simulate('change', { target: { value } });

    const form = component.find('form');
    form.simulate('submit', { preventDefault: () => {} });

    expect(props.onSubmit).toHaveBeenCalledWith(value);
  });
});
