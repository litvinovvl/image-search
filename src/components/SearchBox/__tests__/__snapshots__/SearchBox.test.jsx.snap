// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render SearchBox properly 1`] = `
<form
  className="search-wrapper"
  onSubmit={[Function]}
>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="What images would ypu like to see on Pixabay?"
  />
  <button
    className="search-btn"
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`App should render SearchBox properly when images loaded 1`] = `
<form
  className="search-wrapper shrink"
  onSubmit={[Function]}
>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="What images would ypu like to see on Pixabay?"
  />
  <button
    className="search-btn"
    type="submit"
  >
    Search
  </button>
</form>
`;
